scripts/matterircd_complete.pl

```diff
@@ -0,0 +1,815 @@
+#
+# For the full matterircd_complete experience, your matterircd.toml
+# should have SuffixContext=true, ThreadContext="mattermost", and
+# Unicode=true.
+#
+# Add it to ~/.irssi/scripts/autorun, or:
+#
+#   /script load ~/.irssi/scripts/matterircd_complete.pl
+#   /set matterircd_complete_networks <...>
+#
+# NOTE: It is important to set which networks to enable plugin for per
+# above ^.
+#
+# Bind message/thread ID completion to a key to make it easier to
+# reply to threads:
+#
+#   /bind ^G /message_thread_id_search
+#
+# Also bind to insert nicknames:
+#
+#   /bind ^F /nicknames_search
+#
+# (Or pick your own shortcut keys to bind to).
+#
+# Then:
+#
+#   Ctrl+g - Insert latest message/thread ID.
+#   Ctrl+c - Abort inserting message/thread ID. Also clears existing.
+#
+#   @@+TAB to tab auto-complete message/thread ID.
+#   @ +TAB to tab auto-complete IRC nick. Active users appear first.
+#
+# By default, message/thread IDs are shortened from 26 characters to
+# first few (default 5). It is also grayed out to try reduce noise and
+# make it easier to read conversations. To disable this use:
+#
+#   /set matterircd_complete_shorten_message_thread_id 0
+#
+# Use the dump commands to show the contents of the cache:
+#
+#   /matterircd_complete_msgthreadid_cache_dump
+#   /matterircd_complete_nick_cache_dump
+#
+# (You can bind these to keys).
+#
+# To increase or decrease the size of the cache, use:
+#
+#   /set matterircd_complete_message_thread_id_cache_size 50
+#   /set matterircd_complete_nick_cache_size 20
+#
+# To ignore specific nicks in autocomplete:
+#
+#   /set matterircd_complete_nick_ignore somebot anotherbot
+
+use strict;
+use warnings;
+use experimental 'smartmatch';
```

Comment:
if you want, you could `use List::Util 'any';` and `any { $_ eq $needle } @array ` instead of `~~`